node/resolvers/Queries/Organizations.ts

```diff
@@ -241,6 +241,111 @@ const Organizations = {
     }
   },
 
+  getOrganizationsByEmailPaginated: async (
+    _: void,
+    {
+      email,
+      page = 1,
+      pageSize = 25,
+    }: {
+      email: string
+      page: number
+      pageSize: number
+    },
+    {
+      clients: { storefrontPermissions, session },
+      vtex: { logger, sessionToken, adminUserAuthToken },
+    }: any
+  ) => {
+    const organizationFilters: string[] = []
+    let fromSession = false
+    const {
+      data: { checkUserPermission },
+    }: any = await storefrontPermissions
+      .checkUserPermission('vtex.b2b-organizations@1.x')
+      .catch((error: any) => {
+        logger.error({
+          error,
+          message: 'checkUserPermission-error',
+        })
+
+        return {
+          data: {
+            checkUserPermission: null,
+          },
+        }
+      })
+
+    if (
+      (!adminUserAuthToken &&
+        !checkUserPermission?.permissions.includes('add-sales-users-all')) ||
+      !(email?.length > 0)
+    ) {
+      const sessionData = await session
+        .getSession(sessionToken as string, ['*'])
+        .then((currentSession: any) => {
+          return currentSession.sessionData
+        })
+        .catch((error: any) => {
+          logger.warn({
+            error,
+            message: 'getOrganizationsByEmail-session-error',
+          })
+
+          return null
+        })
+
+      if (checkUserPermission?.permissions.includes('add-users-organization')) {
+        const orgId =
+          sessionData?.namespaces?.['storefront-permissions']?.organization
+            ?.value
+
+        if (!orgId) {
+          throw new Error('No permission for getting the organizations')
+        }
+
+        organizationFilters.push(orgId)
```

Comment:
session will have just one org id, the current logged org, right? and below we are filtering the result to return only the orgs added to `organizationsFilters`, so the `getOrganizationsByEmail` would not return all the organizations for the given email in this case, right?

Reply:
Hi @enzomerca 
Is the same approach of the getOrganizationsByEmail query. This is the case for the request from the storefront where there's no admin auth cookie.

Reply:
@vhaalmeida We would like to avoid old mistakes, and we have a lot of tasks to improve the source code and the B2B suite app implementations. So our advice is to adjust the rules to test.